pkg/foxtrot/db.go

```diff
@@ -0,0 +1,70 @@
+package foxtrot
+
+import (
+	"context"
+	"database/sql"
+	"errors"
+	"fmt"
+
+	"foxygo.at/s/errs"
+)
+
+var errDBInitialisation = errors.New("database initialisation error")
+
+type db struct {
+	conn *sql.DB
+}
+
+func newDB(dsnURI string) (*db, error) {
+	conn, err := sql.Open("sqlite3", dsnURI)
+	if err != nil {
+		return nil, errs.New(errDBInitialisation, err)
+	}
+
+	db := &db{conn: conn}
+	if err := db.setupSchema(); err != nil {
+		_ = db.conn.Close()
+		return nil, err
+	}
+
+	return db, nil
+}
+
+func (db *db) close() {
+	_ = db.conn.Close()
+}
+
+func (db *db) setupSchema() error {
+	selectVersionStr := "SELECT version FROM schema"
+	version := ""
+	err := db.conn.QueryRow(selectVersionStr).Scan(&version)
+	expectedVersion := "v0.0.1"
+	if err == nil && version == expectedVersion {
+		return nil
+	} else if err == nil {
```

Comment:
perhaps swap the two branches of the `if`:
```go
if err == nil && version != expectedVersion {
    return errs...
} else if err == nil {
    return nil
}

// Assume an error means the schema has not been created, so create it now.
```

I think that makes the error condition a little clearer. Also that comment makes it a little easier to understand what is going on I think.

Reply:
YES! thanks.